dialogue-core/src/main/java/com/palantir/dialogue/core/RoundRobinChannel.java

```diff
@@ -46,11 +50,14 @@
         int host = currentHost.getAndUpdate(value -> toIndex(value + 1));
 
         for (int i = 0; i < delegates.size(); i++) {
-            LimitedChannel channel = delegates.get(toIndex(host + i));
-            Optional<ListenableFuture<Response>> maybeCall = channel.maybeExecute(endpoint, request);
-            if (maybeCall.isPresent()) {
-                return maybeCall;
+            CompositeLimitedChannel channel = delegates.get(toIndex(host + i));
+            LimitedResponse maybeCall = channel.maybeExecute(endpoint, request);
+            if (maybeCall.matches(isBlacklisted)) {
+                continue;
+            } else if (maybeCall.matches(isLimited)) {
+                return Optional.empty();
```

Comment:
If a response is limited by one channel (too many concurrent requests?) we don't attempt to rotate to the next channel?

Reply:
ya it seems correct to skip over the the limited channel

Reply:
But this stops iterating and doesn't attempt any more channels unless I'm reading it incorrectly.

Reply:
Yes, I think its right to stop iterating if you hit a limited channel to give time for the limit to be released. Otherwise you'll end up just spinning until a limit is released

Reply:
Each delegate has its own limit, the next channel may not be limited in the same way

Reply:
For example when a new host is discovered we want to allow most requests to rotate to that channel even if the rest are some combination of saturated/blacklisted.